flink/v1.19/build.gradle

```diff
@@ -68,6 +68,9 @@ project(":iceberg-flink:iceberg-flink-${flinkMajorVersion}") {
 
     implementation libs.datasketches
 
+    // for caching in DynamicSink
+    implementation libs.caffeine
+
```

Comment:
I added this dependency which was already added in #13247, but is only strictly necessary in this PR. This avoids a dependency of this PR on #13247.